### ISSUE TITLE
Accessibility:  Improving the web accessibility for Jupyterhub page

### ISSUE THREAD

**Issue by alexanderchosen** [jupyterhub/outreachy#125]
### What page is this for?

https://jupyter.org/hub

### WAVE accessibility report

![hub1](https://user-images.githubusercontent.com/56475753/197549209-f2741f8f-d9ce-48d8-a2ea-cdb71eb80c99.png)

![hub2](https://user-images.githubusercontent.com/56475753/197549249-8a6c04d5-366d-4713-bc88-20ccd7629cae.png)

**The following are the relevant accessibility report gotten from WAVE**:

1. **1 alert**: Alerts are used to show elements that may cause accessibility issues and the impact is always decided by the Evaluator. 
A missing first-level heading (h1) was detected by the WAVE tool. This implies that the page doesn’t have a first-level heading which should be present on nearly all pages as headings are very useful for users of many assistive technologies.
The first level heading tag should contain the most important heading on the page if the page presents a main heading.

2. **4 Features**: The features issue for this page indicates 2X alternative text, 1X linked image with alternative text, and 1X language.  
The 2X alternative text points at the alternative text for the Jupyterhub logo at the top and the Project Jupyter logo in the footer below. 
The 1X linked image with alternative text points at the alternative text for the Jupyter logo inside the nav bar with the link to the home page.
1X language identifies the language of the page and allows screen readers to read the content appropriately.

3. **13 Structural elements**: The structural elements evaluate the overall structure of the page. The structural elements are broken into 7X heading level 2, 4X unordered list, 1X navigation, and 1X footer.
7X heading level 2 indicates the use of seven( 7) h2 tags in the page. Also, the 4X unordered list indicates four (4) unordered list types used in the page. 
Furthermore, the 1X Navigation indicates the presence of one(1) nav element in the page. Also, the 1X footer indicates the presence of one (1) footer element in the page.

4. **7 Aria**: Aria presents important accessibility information that must be carefully used. It reduces accessibility when used incorrectly.
The 7X aria label indicates that seven (7) aria labels are present in the page.

### The HTML element to be changed

```html
<!-- Missing H1 element -->
<!-- h1 tag to be added just below the jupyterhub's logo and the text inside the paragraph tag shortened -->

<header class="page-header">
Alternative text
<img src="assets/homepage/hublogo.svg" alt="JupyterHub logo" loading="eager">
<!-- add h1 heading element here -->
<p>
A multi-user version of the notebook designed for companies, classrooms, and research labs
</p>

<!-- unnecessary use of h2 tag in footer section -->
<h2 id="project-jupyter">
Project Jupyter
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#project-jupyter" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h2>

<h2 id="subprojects">
Subprojects
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#subprojects" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h2>

<h2 id="follow-us">
Follow us
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#follow-us" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h2>
```

### Your proposed HTML (or CSS), after change

```html
<!-- Missing h1 element -->
<!-- h1 heading tag plays a major role in search engine optimization, conveys important keywords and screen readers prefer the use of h1 tag as it improves accessibility -->
<!-- I formed a short and catchy phrase for the h1 heading -->

<header class="page-header">
Alternative text
<img src="assets/homepage/hublogo.svg" alt="JupyterHub logo" loading="eager">

<!-- add h1 heading element here -->

<h1> Interactive Data Science and M.L Notebook </h1>

<!-- I shortened the text by removing 'of the notebook' since the h1 tag already talks of a notebook -->

<p>
A multi-user version designed for companies, classrooms, and research labs
</p>

<!-- unnecessary use of h2 tag in footer section -->
<!-- the h2 tag used in the footer section could be replaced with the h4 tag. h2 tags are 1.5em (24 pixels) in size and that's not the required text size. h4 tags are 1em (16 pixels) and it fits works on most browsers-->

<h4 id="project-jupyter">
Project Jupyter
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#project-jupyter" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h4>

<h4 id="subprojects">
Subprojects
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#subprojects" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h4>

<h4 id="follow-us">
Follow us
ARIA label
<a class="anchorjs-link keychainify-checked" aria-label="Anchor" data-anchorjs-icon="" href="#follow-us" style="font: 1em / 1 anchorjs-icons; padding-left: 0.375em;"></a>
</h4>
```

### MY SUGGESTIONS

The JupyterHub page is structured like a newspaper with **no catchy user interface that leaves users or readers with a special feel or experience**. Looking at the page from a first-timer perspective, it has much text without images and designs to better illustrate the topic.

I suggest that the webpage **should be re-designed with a better user interface to give users a better reading feel and improve the pages accessibility**. The following sub-topics should be better illustrated using div cards to convey different information, links to official pages, articles, and videos.

- Why JupyterHub?
- Latest tools and Features
- How to Guide
- Community Forums

Each sub-topic should be highlighted with beautiful UI designs and visible links to other useful and relevant resources.

```

**Comment by alexanderchosen** [comment]
@sgibson91 @minrk I properly reviewed JupyterHub's webpage using the WAVE tool and I also noted some areas which need changes. 
I look forward to your feedback and corrections. Thanks

**Comment by minrk** [comment]
Thank you for the suggestions!

**Comment by alexanderchosen** [comment]
It's was nice working on this issue and i learnt alot of new things especially on how to properly evaluate a webpage accessibility. 
Thanks alot @minrk

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
